iothub/service/src/Common/Exceptions/IotHubException.cs

```diff
@@ -47,11 +47,23 @@ public IotHubException(string message, Exception innerException)
         {
         }
 
+        public IotHubException(ErrorCode code, string message, Exception innerException = null)
+            : this(message, innerException, false, string.Empty)
+        {
```

Comment:
What do you think about having the ErrorCode overload on every constructor?

Reply:
Would that not be a breaking change as all these classes are public?

Reply:
We would be adding additional APIs, so no, it would be a public API surface change, but not breaking.

Reply:
I thought about this as well. I can definitely add the ErrorCode on all the constructors. I added it for a single contructor based on who needs to consume it. For me it was how much time and effort do we want to put in for this bug fix, since track2 is going to replace this one. I am not opposed to the idea of adding ErrorCode to all constructors on all the exception types we implement.

Reply:
@abhipsaMisra  - I misunderstood what you meant. Sure adding a new constructor overload will not be a breaking change. But I agree with @barustum that it is probably not worth spending the time on it. This is mostly consumed internally so this might be sufficient. It could be another improvement story on the board that can be prioritized accordingly.

Reply:
that works for me